pkg/codegen/python/gen.go

```diff
@@ -2644,9 +2706,61 @@ func (mod *modContext) genType(w io.Writer, name, comment string, properties []*
 
 	// Generate properties. Input types have getters and setters, output types only have getters.
 	mod.genProperties(w, props, input /*setters*/, "", func(prop *schema.Property) string {
-		return mod.typeString(prop.Type, input, false /*acceptMapping*/)
+		return mod.typeString(prop.Type, input, false /*acceptMapping*/, false /*forDict*/)
+	})
+
+	fmt.Fprintf(w, "\n")
+	return nil
+}
+
+func (mod *modContext) genDictType(w io.Writer, name, comment string, properties []*schema.Property) error {
+	// Sort required props first.
+	props := make([]*schema.Property, len(properties))
+	copy(props, properties)
+	sort.Slice(props, func(i, j int) bool {
+		pi, pj := props[i], props[j]
+		switch {
+		case pi.IsRequired() != pj.IsRequired():
+			return pi.IsRequired() && !pj.IsRequired()
+		default:
+			return pi.Name < pj.Name
+		}
 	})
 
+	indent := "    "
+	name = pythonCase(name)
+
+	// Running mypy gets very slow when there are a lot of TypedDicts.
```

Comment:
Should this be a TODO with an open issue on our side linking to the upstream issue?  Assuming it would be better to at some point also run this for mypy if the upstream issue is fixed?

Reply:
Good idea! I created https://github.com/pulumi/pulumi/issues/16408 and linked via TODO